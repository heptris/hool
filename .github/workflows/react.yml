name: React CI

on:
  push:
    branches:
      - develop

jobs:
  build:
    runs-on: ubuntu-20.04

    steps:
      - uses: actions/checkout@v3

      - name: Create secrets
        run: echo "${{ secrets.FRONT_DOTENV_LOCAL }}" > frontend/.env.local

      - name: Set up Node.js 16.16
        uses: actions/setup-node@v3
        with:
          node-version: 16.16

      - name: Install dependencies
        run: |
          cd frontend/
          npm install

      # - name: Run tests
      #   run: |
      #     cd frontend/
      #     npm run test

      - name: Dockerize
        run: |
          echo "${{ secrets.FRONT_DOCKER_HUB_PW }}" | docker login -u "${{ secrets.FRONT_DOCKER_HUB_USER }}" --password-stdin
          docker build -t "${{ secrets.FRONT_IMAGE_NAME }}" ./frontend/
          docker push "${{ secrets.FRONT_IMAGE_NAME }}"
          docker logout
