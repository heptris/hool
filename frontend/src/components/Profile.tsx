import { useState } from "react";

import styled from "styled-components";
import { darkTheme } from "../styles/Theme";

import profileDefaultImg from "../assets/profile-default-img.png";

const Container = styled.div`
  width: 100vw;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
`;

const ProfileBox = styled.div`
  border: 1px solid ${darkTheme.adaptiveGrey700};
  border-radius: 4px;
  width: 20rem;
  background-color: ${darkTheme.mainColor};
  padding: 0.8rem;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const ProfileHeader = styled.span`
  align-self: start;
  font-size: 1.25rem;
  font-weight: bold;
`;

const ProfileImg = styled.img`
  width: 5rem;
  border-radius: 3rem;
  margin-top: 3rem;
`;

const Username = styled.span`
  margin-top: 2rem;
  font-size: 1.25rem;
  font-weight: bold;
`;

const Email = styled.span`
  color: ${darkTheme.adaptiveGrey200};
  font-size: 0.875rem;
  margin: 0.5rem 0 0 0;
`;

const InfoBox = styled.div`
  display: flex;
  width: 70%;
  justify-content: space-between;
  margin: 6rem 0 3rem 0;
`;

const Info = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;
`;

const InfoContent = styled.span`
  color: ${darkTheme.adaptiveGrey200};
  font-size: 0.875rem;
  margin: 1rem 0 0 0;
`;

const InfoNumber = styled.span`
  font-size: 1.25rem;
  font-weight: bold;
`;

function Profile() {
  const [username, setUsername] = useState<string | null>("Andrew");
  const [email, setEmail] = useState<string | null>("Andy@gmail.com");

  const [point, emojiCnt, followingCnt]: number[] = [1237, 2, 47];
  const [userPoint, setUserPoint] = useState<string | null>(
    point?.toLocaleString("ko-KR")
  );
  const [userEmojiCnt, setUserEmojiCnt] = useState<string | null>(
    emojiCnt?.toLocaleString("ko-KR")
  );
  const [userFollwingCnt, setUserFollowingCnt] = useState<string | null>(
    followingCnt?.toLocaleString("ko-KR")
  );

  return (
    <Container>
      <ProfileBox>
        <ProfileHeader>나의 프로필</ProfileHeader>

        <ProfileImg src={profileDefaultImg} alt="profile-img" />

        <Username>{username}</Username>
        <Email>{email}</Email>

        <InfoBox>
          <Info>
            <InfoNumber>{userPoint}</InfoNumber>
            <InfoContent>포인트</InfoContent>
          </Info>

          <Info>
            <InfoNumber>{userEmojiCnt}</InfoNumber>
            <InfoContent>이모지</InfoContent>
          </Info>

          <Info>
            <InfoNumber>{userFollwingCnt}</InfoNumber>
            <InfoContent>팔로잉</InfoContent>
          </Info>
        </InfoBox>
      </ProfileBox>
    </Container>
  );
}

export default Profile;
