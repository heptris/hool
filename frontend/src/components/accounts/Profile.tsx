import { useState } from "react";

import styled from "styled-components";
import { darkTheme } from "styles/Theme";

import profileDefaultImg from "assets/profile-default-imgs/1.png";

import Button from "../commons/Button";
import Modal from "../commons/Modal";
import Card from "../commons/Card";

function Profile() {
  const [isEditing, setIsEditing] = useState(false);
  const [isDisplayModal, setIsDisplayModal] = useState(false);

  const switchIsEditing = () => {
    setIsEditing(!isEditing);
  };

  const switchIsDisplayModal = () => {
    setIsDisplayModal(!isDisplayModal);
  };

  const [username, setUsername] = useState<string | null>("Andrew");
  const [email, setEmail] = useState<string | null>("Andy@gmail.com");

  const [point, emojiCnt, followingCnt]: number[] = [1237, 2, 47];
  const [userPoint, setUserPoint] = useState<string | null>(
    point?.toLocaleString("ko-KR")
  );
  const [userEmojiCnt, setUserEmojiCnt] = useState<string | null>(
    emojiCnt?.toLocaleString("ko-KR")
  );
  const [userFollwingCnt, setUserFollowingCnt] = useState<string | null>(
    followingCnt?.toLocaleString("ko-KR")
  );

  return (
    <>
      {isDisplayModal && (
        <Modal
          header={
            <PointLogHeader>
              <span>큐브 획득 내역</span>
            </PointLogHeader>
          }
          body={
            <PointLogBody>
              <Card marginTop={1}>
                <CardContent>
                  <div>
                    <Activity>로그인 보너스</Activity>
                    <Date>2022.07.25</Date>
                  </div>
                  <Diff>+500</Diff>
                </CardContent>
              </Card>
            </PointLogBody>
          }
          onDisplayChange={switchIsDisplayModal}
        />
      )}

      <ProfileBox>
        <ProfileHeader>나의 프로필</ProfileHeader>

        <ProfileImg src={profileDefaultImg} alt="profile-img" />

        <Username>{username}</Username>
        <Email>{email}</Email>

        <div onClick={switchIsEditing}>
          <Button width={6} height={2} text={"프로필 편집"} />
        </div>

        <InfoBox>
          <Clickable onClick={switchIsDisplayModal}>
            <Info>
              <InfoNumber>{userPoint}</InfoNumber>
              <InfoContent>큐브</InfoContent>
            </Info>
          </Clickable>

          <Info>
            <InfoNumber>{userEmojiCnt}</InfoNumber>
            <InfoContent>이모지</InfoContent>
          </Info>

          <Info>
            <InfoNumber>{userFollwingCnt}</InfoNumber>
            <InfoContent>친구</InfoContent>
          </Info>
        </InfoBox>
      </ProfileBox>
    </>
  );
}

const ProfileBox = styled.div`
  border: 1px solid ${darkTheme.adaptiveGrey700};
  border-radius: 4px;
  width: 30rem;
  background-color: ${darkTheme.mainColor};
  padding: 0.8rem;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const ProfileHeader = styled.span`
  align-self: start;
  font-size: 1.25rem;
  font-weight: bold;
`;

const ProfileImg = styled.img`
  width: 5rem;
  border-radius: 3rem;
  margin-top: 3rem;
`;

const Username = styled.span`
  margin-top: 2rem;
  font-size: 1.25rem;
  font-weight: bold;
`;

const Email = styled.span`
  color: ${darkTheme.adaptiveGrey200};
  font-size: 0.875rem;
  margin: 0.5rem 0 2.5rem 0;
`;

const InfoBox = styled.div`
  display: flex;
  width: 80%;
  justify-content: space-between;
  margin: 5rem 0 3rem 0;
`;

const Info = styled.div`
  display: flex;
  flex-direction: column;
  text-align: center;
  padding: 1rem;
  width: 3rem;
`;

const Clickable = styled.div`
  &:hover {
    cursor: pointer;
    background-color: ${darkTheme.darkColor};
    border-radius: 8px;
  }
`;

const InfoContent = styled.span`
  color: ${darkTheme.adaptiveGrey200};
  font-size: 0.875rem;
  margin: 1rem 0 0 0;
`;

const InfoNumber = styled.span`
  font-size: 1.25rem;
  font-weight: bold;
`;

const PointLogHeader = styled.h1`
  width: 30rem;
  margin: 1rem 0 1rem 0;
`;

const PointLogBody = styled.div`
  width: 100%;
  height: 30rem;
  overflow: auto;

  ::-webkit-scrollbar {
    width: 6px;
  }
  ::-webkit-scrollbar-track {
    background-color: transparent;
  }
  ::-webkit-scrollbar-thumb {
    border-radius: 3px;
    background-color: ${darkTheme.adaptiveGrey500};
  }
  ::-webkit-scrollbar-button {
    width: 0;
    height: 0;
  }
`;

const CardContent = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 1rem 1.5rem;
`;

const Activity = styled.h2`
  font-size: 1.25rem;
  margin: 0 0 0.5rem 0;
`;

const Date = styled.span`
  font-size: 0.875rem;
  color: ${darkTheme.adaptiveGrey500};
`;

const Diff = styled.span`
  font-size: 1.25rem;
  font-weight: bold;
`;

export default Profile;
