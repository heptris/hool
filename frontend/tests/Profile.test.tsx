import React from "react";
import { render } from "@testing-library/react";
import Profile from "../src/components/accounts/Profile";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const utils = render(<Profile />);
    expect(utils.container).toMatchSnapshot();
  });

  it("shows the props correctly", () => {
    const utils = render(<Profile />);
    utils.getByText("나의 프로필");
    utils.getByText("큐브");
    utils.getByText("이모지");
    utils.getByText("친구");
  });
});
