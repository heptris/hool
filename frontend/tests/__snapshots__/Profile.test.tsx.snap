// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div
    class="sc-ftvSup bxhYay"
  >
    <span
      class="sc-papXJ hKCaeh"
    >
      나의 프로필
    </span>
    <img
      alt="profile-img"
      class="sc-jqUVSM bpIlDa"
      src="default"
    />
    <span
      class="sc-kDDrLX lbmtVj"
    >
      Andrew
    </span>
    <span
      class="sc-iqcoie lhOjwE"
    >
      Andy@gmail.com
    </span>
    <div>
      <button
        class="sc-bczRLJ dtRITV"
        height="2"
        width="6"
      >
        <span>
          프로필 편집
        </span>
      </button>
    </div>
    <div
      class="sc-crXcEl jWdfjH"
    >
      <div
        class="sc-breuTD dnDSjI"
      >
        <div
          class="sc-evZas gcCZdA"
        >
          <span
            class="sc-hAZoDl bgMyON"
          >
            1,237
          </span>
          <span
            class="sc-ksZaOG iMNuZq"
          >
            포인트
          </span>
        </div>
      </div>
      <div
        class="sc-evZas gcCZdA"
      >
        <span
          class="sc-hAZoDl bgMyON"
        >
          2
        </span>
        <span
          class="sc-ksZaOG iMNuZq"
        >
          이모지
        </span>
      </div>
      <div
        class="sc-evZas gcCZdA"
      >
        <span
          class="sc-hAZoDl bgMyON"
        >
          47
        </span>
        <span
          class="sc-ksZaOG iMNuZq"
        >
          친구
        </span>
      </div>
    </div>
  </div>
</div>
`;
